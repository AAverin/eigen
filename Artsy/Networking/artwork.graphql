query OldArtworkQuery($artworkID: String!) {
  artwork(id: $artworkID) {
    artist {
      id
      _id
      name
      years
      birthday
      nationality
      blurb

      image {
        url
      }
      sortable_id
    }

    partner {
      name

      id
      default_profile_id
      is_default_profile_public
      type
      href
    }

    id
    _id
    date
    title

    can_share_image
    collecting_institution
    images {
      id
      image_versions
      image_url

    }
    additional_information
    dimensions {
      cm
      in
    }
    attribution_class {
      id
      name
      long_description
      short_description
    	info
    }
    edition_sets {
      id
      sale_message
      is_sold
      is_for_sale
      dimensions {
        in
        cm
      }
    }
    medium
    literature
    provenance
    series
    signature
    price

    category
    exhibition_history
    is_for_sale
    image_rights
    published
    sale_message
    is_sold
    is_price_hidden
  }
}

