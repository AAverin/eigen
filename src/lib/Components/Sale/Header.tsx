import * as React from "react"
import { Dimensions, NativeModules, StyleSheet, Text, TextStyle, View, ViewProperties, ViewStyle } from "react-native"
import { createFragmentContainer, graphql } from "react-relay"
import OpaqueImageView from "../OpaqueImageView"
import Headline from "../Text/Headline"

class Header extends React.Component<RelayProps, {}> {
  constructor(props) {
    super(props)
    this.state = {}
  }

  render() {
    const { width, height } = Dimensions.get("window")
    const sale = this.props.sale
    return (
      <View>
        <View style={styles.header}>
          {this.props.showImage
            ? <OpaqueImageView
                // TODO: un-hardcode this (metaphysics isn't resolving sales' cover_image field property.)
                imageURL="https://d32dm0rphc51dk.cloudfront.net/3YueUJ2y1Vwx-YQJ82lG_w/wide.jpg"
                style={{ position: "absolute", top: 0, width, height }}
              />
            : null}
          <Headline style={styles.headline} numberOfLines={2}>
            {sale.name}
          </Headline>
        </View>
      </View>
    )
  }
}

export default createFragmentContainer(
  Header,
  graphql`
    fragment Header_sale on Sale {
      cover_image {
        href
      }
    }
  `
)

const isPad = Dimensions.get("window").width > 700

interface Styles {
  header: TextStyle
  headline: TextStyle
}

const styles = StyleSheet.create<Styles>({
  header: {
    marginTop: 15,
    height: 36,
    justifyContent: "center",
    alignItems: "center",
    paddingLeft: 40,
    paddingRight: 40,
  },
  headline: {
    textAlign: "center",
    fontSize: isPad ? 20 : 14,
  },
})

interface RelayProps {
  sale: {
    _id: string
    id: string
    name: string | null
  }
  showImage: boolean
}
