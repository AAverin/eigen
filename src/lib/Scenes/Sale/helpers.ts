import moment from "moment"

export type SaleStatus = "notYetOpen" | "active" | "closed"

export const isCurrentlyActive = (startAt: string, endAt: string) => {
  const now = moment()
  return moment(startAt) < now && now < moment(endAt)
}

export const saleStatus = (startAt: string | null, endAt: string | null): SaleStatus => {
  if (startAt === null || endAt === null) {
    return "notYetOpen"
  }

  if (isCurrentlyActive(startAt, endAt)) {
    return "active"
  }

  const now = moment()
  if (moment(startAt) > now) {
    return "notYetOpen"
  }

  return "closed"
}
