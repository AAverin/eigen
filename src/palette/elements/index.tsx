export * from "./Cards"
export * from "./Touchable"
export * from "./Typography"
