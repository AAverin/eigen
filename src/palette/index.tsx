export * from "./atoms"
export * from "./organisms"
export * from "./elements"
export * from "./Theme"
