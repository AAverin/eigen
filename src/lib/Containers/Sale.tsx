import * as React from "react"
import { Dimensions, StyleSheet, View, ViewProperties, ViewStyle } from "react-native"
import ParallaxScrollView from "react-native-parallax-scroll-view"
import { createRefetchContainer, graphql, RelayRefetchProp } from "react-relay"

import { GhostButton } from "../Components/Buttons"
import Header from "../Components/Sale/Header"
import SaleArtworksGrid from "../Components/ArtworkGrids/RelayConnections/SaleArtworksGrid"
import Separator from "../Components/Separator"
import SerifText from "../Components/Text/Serif"

const isPad = Dimensions.get("window").width > 700
const HeaderHeight = 64

interface Props extends ViewProperties {
  sale: any
}

interface State {
  showingStickyHeader: boolean
}

export class Sale extends React.Component<Props, State> {
  foregroundHeight: number = 200

  constructor(props: Props) {
    super(props)
    this.state = {
      showingStickyHeader: true,
    }
  }

  /** Top of the Component */
  renderForeground = () => {
    const containerStyle = {
      backgroundColor: "white",
      paddingLeft: this.commonPadding,
      paddingRight: this.commonPadding,
    }
    return (
      <View style={[containerStyle, styles.header]}>
        <Header sale={this.props.sale} />
      </View>
    )
  }

  renderStickyRefineSection() {
    return <View />
    // console.log(this.state)
    // const topMargin = this.state.showingStickyHeader ? 0 : HeaderHeight
    // const separatorColor = this.state.showingStickyHeader ? "white" : colors["gray-regular"]

    // const refineButtonWidth = 80
    // const maxLabelWidth = Dimensions.get("window").width - this.commonPadding * 2 - refineButtonWidth - 10

    // return (
    //   <View style={{ backgroundColor: "white" }}>
    //     <Separator style={{ marginTop: topMargin, backgroundColor: separatorColor }} />
    //     <View style={[styles.refineContainer, { paddingLeft: this.commonPadding, paddingRight: this.commonPadding }]}>
    //       <SerifText style={{ fontStyle: "italic", marginTop: 2, maxWidth: maxLabelWidth }}>Hey replace me!</SerifText>
    //       <GhostButton text="REFINE" style={{ height: 26, width: refineButtonWidth }} onPress={this.refineTapped} />
    //     </View>
    //     <Separator style={{ backgroundColor: separatorColor }} />
    //   </View>
    // )
  }

  renderStickyHeader = () => {
    const commonPadding = this.commonPadding
    return (
      <View style={{ paddingLeft: commonPadding, paddingRight: commonPadding, backgroundColor: "white" }}>
        <Header sale={this.props.sale} />
      </View>
    )
  }

  /** Callback from the parallax that we have transitioned into the small title mode */
  onChangeHeaderVisibility = (sticky: boolean) => {
    if (this.state.showingStickyHeader !== sticky) {
      // Set the state so we can change the margins on the refine section
      this.setState({ showingStickyHeader: sticky })
    }
  }

  refineTapped = button => {
    // Aww yeah
  }

  get commonPadding(): number {
    return isPad ? 40 : 20
  }

  render() {
    const stickyTopMargin = this.state.showingStickyHeader ? 0 : -HeaderHeight

    return (
      <ParallaxScrollView
        scrollsToTop={true}
        fadeOutForeground={false}
        backgroundScrollSpeed={1}
        backgroundColor="white"
        contentBackgroundColor="white"
        renderForeground={this.renderForeground}
        stickyHeaderHeight={HeaderHeight}
        renderStickyHeader={this.renderStickyHeader}
        onChangeHeaderVisibility={this.onChangeHeaderVisibility}
        stickyHeaderIndices={[1]}
        renderBodyComponentHeader={this.renderStickyRefineSection}
        parallaxHeaderHeight={this.foregroundHeight}
        parallaxHeaderContainerStyles={{ marginBottom: stickyTopMargin }}
      >
        <View style={{ marginTop: 20, paddingLeft: this.commonPadding, paddingRight: this.commonPadding }}>
          <SaleArtworksGrid sale={this.props.sale} queryKey="gene.artworks" />
        </View>
      </ParallaxScrollView>
    )
  }
}

interface Styles {
  header: ViewStyle
  stickyHeader: ViewStyle
  refineContainer: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  header: {
    width: isPad ? 330 : null,
    alignSelf: isPad ? "center" : null,
  },
  stickyHeader: {
    flexDirection: "row",
    justifyContent: "space-between",
    height: 26,
    marginTop: 12,
    marginBottom: 12,
    paddingLeft: isPad ? 40 : 20,
    paddingRight: isPad ? 40 : 20,
  },
  refineContainer: {
    flexDirection: "row",
    justifyContent: "space-between",
    height: 26,
    marginTop: 12,
    marginBottom: 12,
  },
})

export default createRefetchContainer(
  Sale,
  graphql.experimental`
    fragment Sale_sale on Sale {
      id
      name
      ...Header_sale
      ...SaleArtworksGrid_saleArtworks
    }
  `,
  graphql.experimental`
    query SaleRefetchQuery($saleID: String!) {
      sale(id: $saleID) {
        ...Sale_sale
      }
    }
  `
)
