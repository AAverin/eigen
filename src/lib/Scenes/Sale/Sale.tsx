import { Sale_sale$key } from "__generated__/Sale_sale.graphql"
import { SaleQueryRendererQuery } from "__generated__/SaleQueryRendererQuery.graphql"
import Spinner from "lib/Components/Spinner"
import { Flex, Sans, Spacer, Text } from "palette"
import React from "react"
import { graphql } from "react-relay"
import { useFragment, useQuery } from "relay-hooks"
import { RegisterToBidButton } from "./Components/RegisterToBidButton"

interface Props {
  sale: Sale_sale$key
}

export const Sale: React.FC<Props> = (props) => {
  const sale = useFragment(SaleFragmentSpec, props.sale)

  return (
    <Flex mx="3" my="3">
      <Spacer mt={80} />
      <Sans size="4t">Sale name: {sale.name}</Sans>
      <Text selectable>Sale id: {sale.internalID}</Text>
      <RegisterToBidButton sale={sale} />
    </Flex>
  )
}

const SaleFragmentSpec = graphql`
  fragment Sale_sale on Sale {
    name
    internalID
    ...RegisterToBidButton_sale
  }
`

const Placeholder = () => <Spinner style={{ flex: 1 }} />

export const SaleQueryRenderer: React.FC<{ saleID: string }> = ({ saleID }) => {
  const { props, error } = useQuery<SaleQueryRendererQuery>(
    graphql`
      query SaleQueryRendererQuery($saleID: String!) {
        sale(id: $saleID) {
          ...Sale_sale
        }
      }
    `,
    { saleID }
  )
  if (props) {
    return <Sale sale={props.sale!} />
  }
  if (error) {
    throw error
  }

  return <Placeholder />
}
