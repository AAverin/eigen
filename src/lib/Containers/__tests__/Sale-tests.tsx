import React from "react"
import * as renderer from "react-test-renderer"

// Stub out these views for simplicity sake
jest.mock("../../Components/Sale/Header", () => "Header")

import { Sale } from "../Sale"

import artwork from "../../Components/ArtworkGrids/__tests__/__fixtures__/artwork"

import { Theme } from "@artsy/palette"

describe("state", () => {
  it("looks okay by default", () => {
    const sale = {
      saleArtworks: [
        {
          artwork: artwork(),
        },
      ],
    }
    const tree = renderer
      .create(
        <Theme>
          <Sale sale={sale} />
        </Theme>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
