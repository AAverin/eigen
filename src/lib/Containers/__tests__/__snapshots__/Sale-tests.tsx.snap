// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`state looks okay by default 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      undefined,
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "height": 200,
        "opacity": 1,
        "overflow": "hidden",
        "position": "absolute",
        "top": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
          Object {
            "scale": 1,
          },
        ],
        "width": 750,
      }
    }
  >
    <View>
      <View />
    </View>
  </View>
  <RCTScrollView
    onChangeHeaderVisibility={[Function]}
    onScroll={[Function]}
    parallaxHeaderContainerStyles={
      Object {
        "marginBottom": 0,
      }
    }
    scrollEventThrottle={16}
    scrollsToTop={true}
    stickyHeaderIndices={
      Array [
        1,
      ]
    }
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        undefined,
      ]
    }
  >
    <View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "overflow": "hidden",
            },
            Object {
              "marginBottom": 0,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "height": 200,
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              Object {
                "height": 200,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "white",
                    "paddingLeft": 40,
                    "paddingRight": 40,
                  },
                  Object {
                    "alignSelf": "center",
                    "width": 330,
                  },
                ]
              }
            >
              <Header
                sale={
                  Object {
                    "saleArtworks": Array [
                      Object {
                        "artwork": Object {
                          "node": Object {
                            "artists": Array [
                              Object {
                                "name": "Jenny Holzer",
                              },
                            ],
                            "date": "2008",
                            "href": "/artwork/jenny-holzer-talking-politics",
                            "id": "artwork-talking-politics",
                            "image": Object {
                              "aspect_ratio": 1.27,
                              "url": "artsy.net/image-url",
                            },
                            "is_in_auction": false,
                            "partner": Object {
                              "name": "Kukje Gallery",
                            },
                            "sale_message": "Not for sale",
                            "title": "TALKING POLITICS",
                          },
                        },
                      },
                    ],
                  }
                }
                showImage={true}
              />
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "backgroundColor": "white",
            }
          }
        >
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e5e5e5",
                  "height": 1,
                  "marginLeft": -40,
                  "marginRight": -40,
                },
                Object {
                  "backgroundColor": "#e5e5e5",
                  "marginTop": 0,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "height": 26,
                  "justifyContent": "space-between",
                  "marginBottom": 12,
                  "marginTop": 12,
                },
                Object {
                  "paddingLeft": 40,
                  "paddingRight": 40,
                },
              ]
            }
          >
            <Text
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "fontSize": 17,
                  },
                  Object {
                    "fontStyle": "italic",
                    "marginTop": 2,
                    "maxWidth": 580,
                  },
                  Object {
                    "fontFamily": "AGaramondPro-Regular",
                  },
                ]
              }
            >
              Hey replace me!
            </Text>
            <View
              accessible={true}
              flexDirection="row"
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "display": "flex",
                    "flexDirection": "row",
                  },
                ]
              }
            >
              <View
                onPress={[Function]}
                px={1}
                size="small"
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "borderRadius": 3,
                      "borderWidth": 1,
                      "justifyContent": "center",
                      "paddingLeft": 10,
                      "paddingRight": 10,
                      "position": "relative",
                      "width": "auto",
                    },
                    Object {
                      "backgroundColor": "rgba(255, 255, 255, 1)",
                      "borderColor": "rgba(229, 229, 229, 1)",
                      "color": "rgba(0, 0, 0, 1)",
                      "height": 26,
                      "opacity": 1,
                    },
                  ]
                }
                theme={
                  Object {
                    "borders": Array [
                      "1px solid",
                      "2px solid",
                    ],
                    "breakpoints": Array [
                      768,
                      900,
                      1024,
                      1192,
                    ],
                    "colors": Object {
                      "black10": "#E5E5E5",
                      "black100": "#000",
                      "black30": "#C2C2C2",
                      "black5": "#F8F8F8",
                      "black60": "#666",
                      "black80": "#333",
                      "green100": "#0EDA83",
                      "purple100": "#6E1EFF",
                      "purple30": "#D3BBFF",
                      "purple5": "#F8F3FF",
                      "red100": "#F7625A",
                      "white100": "#FFF",
                      "yellow10": "#FDF7E8",
                      "yellow100": "#F1AF1B",
                      "yellow30": "#FAE7BA",
                    },
                    "fontFamily": Object {
                      "display": Object {
                        "regular": "AvantGardeGothicITC",
                      },
                      "sans": Object {
                        "italic": "Unica77LL-Italic",
                        "medium": "Unica77LL-Medium",
                        "mediumItalic": "Unica77LL-MediumItalic",
                        "regular": "Unica77LL-Regular",
                      },
                      "serif": Object {
                        "italic": "ReactNativeAGaramondPro-Italic",
                        "regular": "ReactNativeAGaramondPro-Regular",
                        "semibold": "ReactNativeAGaramondPro-Semibold",
                      },
                    },
                    "grid": Object {
                      "breakpoints": Object {
                        "lg": 1024,
                        "md": 900,
                        "sm": 768,
                        "xl": 1192,
                        "xs": 767,
                      },
                      "col": Object {
                        "padding": 0,
                      },
                      "container": Object {
                        "padding": 0,
                      },
                      "row": Object {
                        "padding": 0,
                      },
                    },
                    "mediaQueries": Object {
                      "hover": "not all and (pointer: coarse), not all and (-moz-touch-enabled: 1)",
                      "lg": "(min-width: 1024px) and (max-width: 1191px)",
                      "md": "(min-width: 900px) and (max-width: 1023px)",
                      "sm": "(min-width: 768px) and (max-width: 899px)",
                      "xl": "(min-width: 1192px)",
                      "xs": "(max-width: 767px)",
                    },
                    "space": Object {
                      "0.3": 3,
                      "0.5": 5,
                      "1": 10,
                      "12": 120,
                      "18": 180,
                      "2": 20,
                      "3": 30,
                      "4": 40,
                      "6": 60,
                      "9": 90,
                    },
                    "typeSizes": Object {
                      "display": Object {
                        "2": Object {
                          "fontSize": 10,
                          "lineHeight": 12,
                        },
                        "3t": Object {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                        "4t": Object {
                          "fontSize": 14,
                          "lineHeight": 18,
                        },
                        "5t": Object {
                          "fontSize": 16,
                          "lineHeight": 20,
                        },
                        "6": Object {
                          "fontSize": 18,
                          "lineHeight": 22,
                        },
                        "8": Object {
                          "fontSize": 22,
                          "lineHeight": 24,
                        },
                      },
                      "sans": Object {
                        "0": Object {
                          "fontSize": 8,
                          "lineHeight": 8,
                        },
                        "1": Object {
                          "fontSize": 10,
                          "lineHeight": 14,
                        },
                        "10": Object {
                          "fontSize": 42,
                          "lineHeight": 50,
                        },
                        "12": Object {
                          "fontSize": 60,
                          "lineHeight": 66,
                        },
                        "14": Object {
                          "fontSize": 80,
                          "lineHeight": 84,
                        },
                        "16": Object {
                          "fontSize": 100,
                          "lineHeight": 104,
                        },
                        "2": Object {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                        "3": Object {
                          "fontSize": 14,
                          "lineHeight": 24,
                        },
                        "3t": Object {
                          "fontSize": 14,
                          "lineHeight": 20,
                        },
                        "4": Object {
                          "fontSize": 16,
                          "lineHeight": 26,
                        },
                        "4t": Object {
                          "fontSize": 16,
                          "lineHeight": 22,
                        },
                        "5": Object {
                          "fontSize": 18,
                          "lineHeight": 30,
                        },
                        "5t": Object {
                          "fontSize": 18,
                          "lineHeight": 26,
                        },
                        "6": Object {
                          "fontSize": 22,
                          "lineHeight": 30,
                        },
                        "8": Object {
                          "fontSize": 28,
                          "lineHeight": 36,
                        },
                      },
                      "serif": Object {
                        "1": Object {
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                        "10": Object {
                          "fontSize": 44,
                          "lineHeight": 50,
                        },
                        "12": Object {
                          "fontSize": 60,
                          "lineHeight": 70,
                        },
                        "2": Object {
                          "fontSize": 14,
                          "lineHeight": 18,
                        },
                        "3": Object {
                          "fontSize": 16,
                          "lineHeight": 24,
                        },
                        "3t": Object {
                          "fontSize": 16,
                          "lineHeight": 20,
                        },
                        "4": Object {
                          "fontSize": 18,
                          "lineHeight": 26,
                        },
                        "4t": Object {
                          "fontSize": 18,
                          "lineHeight": 22,
                        },
                        "5": Object {
                          "fontSize": 22,
                          "lineHeight": 32,
                        },
                        "5t": Object {
                          "fontSize": 22,
                          "lineHeight": 28,
                        },
                        "6": Object {
                          "fontSize": 26,
                          "lineHeight": 32,
                        },
                        "8": Object {
                          "fontSize": 32,
                          "lineHeight": 38,
                        },
                      },
                    },
                  }
                }
                variant="secondaryOutline"
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "height": "100%",
                        "justifyContent": "center",
                        "position": "absolute",
                        "width": "100%",
                      },
                    ]
                  }
                >
                  <Text
                    color="#000"
                    fontFamily="Unica77LL-Medium"
                    fontSize="12px"
                    lineHeight="16px"
                    style={
                      Array [
                        Object {
                          "color": "#000",
                          "fontFamily": "Unica77LL-Medium",
                          "fontSize": 12,
                          "lineHeight": 16,
                        },
                        Object {},
                      ]
                    }
                  >
                    Refine
                  </Text>
                </View>
                <Text
                  fontFamily="Unica77LL-Medium"
                  fontSize="12px"
                  lineHeight="16px"
                  role="presentation"
                  style={
                    Array [
                      Object {
                        "fontFamily": "Unica77LL-Medium",
                        "fontSize": 12,
                        "lineHeight": 16,
                      },
                      Object {
                        "opacity": 0,
                      },
                    ]
                  }
                >
                  Refine
                </Text>
              </View>
            </View>
          </View>
          <View
            style={
              Array [
                Object {
                  "backgroundColor": "#e5e5e5",
                  "height": 1,
                  "marginLeft": -40,
                  "marginRight": -40,
                },
                Object {
                  "backgroundColor": "#e5e5e5",
                },
              ]
            }
          />
        </View>
      </View>
      <View
        onLayout={[Function]}
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      >
        <View
          style={
            Object {
              "marginTop": 20,
              "paddingLeft": 40,
              "paddingRight": 40,
            }
          }
        >
          <ArtworksGrid
            mapConnectionNodeToArtwork={[Function]}
            mapPropsToArtworksConnection={[Function]}
            sale={
              Object {
                "saleArtworks": Array [
                  Object {
                    "artwork": Object {
                      "node": Object {
                        "artists": Array [
                          Object {
                            "name": "Jenny Holzer",
                          },
                        ],
                        "date": "2008",
                        "href": "/artwork/jenny-holzer-talking-politics",
                        "id": "artwork-talking-politics",
                        "image": Object {
                          "aspect_ratio": 1.27,
                          "url": "artsy.net/image-url",
                        },
                        "is_in_auction": false,
                        "partner": Object {
                          "name": "Kukje Gallery",
                        },
                        "sale_message": "Not for sale",
                        "title": "TALKING POLITICS",
                      },
                    },
                  },
                ],
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "white",
          }
        }
      />
    </View>
  </RCTScrollView>
  <View
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "left": 0,
          "overflow": "hidden",
          "position": "absolute",
          "top": 0,
        },
        Object {
          "height": 64,
          "width": 750,
        },
      ]
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "white",
          "height": 64,
          "opacity": 0,
        }
      }
    >
      <View
        style={
          Object {
            "transform": Array [
              Object {
                "translateY": 64,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "white",
              "paddingLeft": 40,
              "paddingRight": 40,
            }
          }
        >
          <Header
            sale={
              Object {
                "saleArtworks": Array [
                  Object {
                    "artwork": Object {
                      "node": Object {
                        "artists": Array [
                          Object {
                            "name": "Jenny Holzer",
                          },
                        ],
                        "date": "2008",
                        "href": "/artwork/jenny-holzer-talking-politics",
                        "id": "artwork-talking-politics",
                        "image": Object {
                          "aspect_ratio": 1.27,
                          "url": "artsy.net/image-url",
                        },
                        "is_in_auction": false,
                        "partner": Object {
                          "name": "Kukje Gallery",
                        },
                        "sale_message": "Not for sale",
                        "title": "TALKING POLITICS",
                      },
                    },
                  },
                ],
              }
            }
            showImage={false}
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
